### PR TITLE
test: test updated node that disables PoX

### PR DESCRIPTION
**Pull request by obycode**
Uses `"blockstack/stacks-blockchain:sip-022-pox-disable-devnet"` for the stacks-node image. This image is based on stacks-blockchain PR #3677.

To run this successfully you'll also need an unreleased version of clarinet, to get the version of `stacks-devnet-js` with support for epoch 2.2.

```
git clone https://github.com/hirosystems/clarinet.git
cd clarinet
git checkout feat/devnet-epoch-2.2
cd components/stacks-devnet-js
npm install
npm run build-darwin-arm64 # for M1 or M2 Mac
```

Once this has built, then build and run the test with:

```
cd stacks-2-1-testing
rm -rf node_modules
yarn install
yarn vitest --run direct-stacking-pox-disable
```

### CODE REVIEW

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    pox_2_activation: 109,
+    epoch_2_2: 120,
+  };
+
+  beforeAll(() => {
+    orchestrator = buildDevnetNetworkOrchestrator(
+      getNetworkIdFromEnv(),
+      timeline,
+      undefined,
+      "blockstack/stacks-blockchain:sip-022-pox-disable-devnet"
```
Comment (friedger) [inline tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:35]: Extract into a constant?

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    let poxInfo = await getPoxInfo(network);
+
+    // Asserts about pox info for better knowledge sharing
+    expect(poxInfo.contract_id).toBe("ST000000000000000000002AMW42H.pox-2");
+    expect(poxInfo.current_cycle.id).toBe(1);
+
+    // Assert that the next cycle has 100m STX locked
+    expect(poxInfo.current_cycle.stacked_ustx).toBe(0);
+    expect(poxInfo.current_cycle.is_pox_active).toBe(false);
```
Comment (friedger) [inline tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:106]: `toBeFalsy`?

File: tests/integration/constants.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ export namespace Constants {
   export const DEVNET_DEFAULT_EPOCH_2_05 = 102;
   export const DEVNET_DEFAULT_EPOCH_2_1 = 106;
   export const DEVNET_DEFAULT_POX_2_ACTIVATION = 110;
+  export const DEVNET_DEFAULT_EPOCH_2_2 = 120;
```
[the remark this answers is not in the document]
Comment (obycode) [reply tests/integration/constants.ts:6]: 120 is the start of cycle 2. Maybe it should instead be sometime in the middle of a cycle, but I don't think it needs to be the first reward block. Realistically 2.1 has been running for a while, and I wanted to let a couple cycles pass before the transition.

[the remark this answers is not in the document]
Comment (friedger) [reply tests/integration/constants.ts:6]: Does 120 or 130 mean the dangerous functions have not yet been called?

**Comment by friedger** [comment]
Maybe a helper asyncExpectPoxActiveToBeFalsy

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+    // Wait for block N+1 where N is the height of the next reward phase
+    await waitForNextRewardPhase(network, orchestrator, 1);
+
+    const blockHeight = timeline.pox_2_activation + 1;
+    const fee = 1000;
+    const cycles = 12;
+
+    // Bob stacks 30m
+    let response = await broadcastStackSTX(
```
[the remark this answers is not in the document]
Comment (obycode) [reply tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:58]: GitHub Copilot is great at doing this refactoring!

File: tests/integration/constants.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (obycode) [reply tests/integration/constants.ts:6]: No, the dangerous functions will have been called for the prepare phase of cycle 2, but we can control what happens before then in the test, preventing it from crashing.

File: tests/integration/helpers.ts [hunk 5]
```diff
@@ -99,6 +116,35 @@ export const getChainInfo = async (
...
 };
 
+export interface AccountInfo {
+  balance: number;
+  locked: number;
+  unlock_height: number;
+  nonce: number;
+}
+
+export const getAccountInfo = async (
```
Comment (friedger) [inline tests/integration/helpers.ts:126]: We have already `expectAccountToBe`

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+      { network, account: Accounts.WALLET_2, fee, nonce: bobNonce++ },
+      {
+        amount: bobInfo.balance - fee,
+        recipient: Accounts.WALLET_3.stxAddress,
+      }
+    );
+    await asyncExpectStacksTransactionSuccess(orchestrator, response.txid);
+
+    // Check Alice's account info
+    const aliceInfo = await getAccountInfo(
```
Comment (friedger) [inline tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:158]: Use
``` 
await expectAccountToBe(network,
      Accounts.WALLET_1.stxAddress,
     100_000_000_000_000 - fee, 
     0)
```

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+    // Verify that Alice's STX are really unlocked by doing a transfer
+    response = await broadcastSTXTransfer(
+      { network, account: Accounts.WALLET_1, fee, nonce: aliceNonce++ },
+      {
+        amount: aliceInfo.balance - fee,
+        recipient: Accounts.WALLET_3.stxAddress,
+      }
+    );
+    await asyncExpectStacksTransactionSuccess(orchestrator, response.txid);
```
Comment (friedger) [inline tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:173]: I like this!

File: tests/integration/helpers.ts (hunk 5 above)
Comment (obycode) [reply tests/integration/helpers.ts:126]: Oh, nice! Oops, I missed that.

Comment (obycode) [reply tests/integration/helpers.ts]: Refactored in 9fca243.

File: tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts (hunk 6 above)
Comment (obycode) [reply tests/integration/pox/stacking/direct-stacking-pox-disable.spec.ts:162]: 9fca243

**Comment by obycode** [comment]
I rushed that last commit and it looks like it's not passing. I'll fix it in a few minutes.

**Comment by obycode** [comment]
Ok resolved.

**Pull request merged**